jwst/ramp_fitting/ramp_fit_step.py

```diff
@@ -173,6 +179,202 @@ def create_optional_results_model(input_model, opt_info):
     return opt_model
 
 
+def compute_RN_variances(groupdq, readnoise_2d, gain_2d, group_time):
+    """
+    Compute the variances due to the readnoise for all integrations.
+
+    Parameters
+    ----------
+    groupdq : ndarray
+        The group data quality array for the exposure, 4-D flag.
+        For groups that have been flagged as both UNDERSAMP and
+        DO_NOT_USE, both flags have been reset.
+
+    readnoise_2d : ndarray
+        readnoise values for all pixels in the image, 2-D float
+
+    gain_2d : ndarray
+        gain values for all pixels in the image, 2-D float
+
+    group_time : float
+        Time increment between groups, in seconds.
+
+    Returns
+    -------
+    var_r2 : ndarray
+        Image of integration-specific values for the slope variance due to
+        readnoise only, 2-D float
+
+    var_r3 : ndarray
+        Cube of integration-specific values for the slope variance due to
+        readnoise only, 3-D float
+
+    var_r4 : ndarray
+        Hypercube of segment- and integration-specific values for the slope
+        variance due to read noise only, 4-D float
+    """
+    nint = groupdq.shape[0]
+    ngroups = groupdq.shape[1]
+    nrows = groupdq.shape[2]
+    ncols = groupdq.shape[3]
```

Comment:
Could simplify these 4 lines to just `nint, ngroups, nrows, ncols = groupdq.shape`.

Reply:
Done